src/io/websocket.js

```diff
@@ -11,10 +11,11 @@ export function streamAll(url, receive) {
     const ws = new ReconnectingWebsocket(url);
     ws.addEventListener("message", e => {
       const obj = JSON.parse(e.data);
-      if (turns[obj.Turn]) {
+      const eventData = obj.Data;
+      if (turns[eventData.Turn]) {
         return;
       }
-      turns[obj.Turn] = true;
+      turns[eventData.Turn] = true;
```

Comment:
Is the idea with this `turns` bit to detect when a duplicate frame is handled? I know you're just preserving it here, but I wonder if it would simplify things a bit if this happened in `onEngineEvent`?

Might be more complicated than I'm thinking, so totally optional suggestion!

Reply:
Yeah, that's what it seems to be doing. I'm preserving it for now. It's rather flaky IMO. I plan to fix this websocket so it isn't tied to a specific value of the message data but didn't want to open that can of worms until later.